src/main/java/useCases/createCorporatePage/PageGenerateInteractor.java

```diff
@@ -0,0 +1,59 @@
+package useCases.createCorporatePage;
+import entity.CorporatePage;
+import entity.Page;
+
+public class PageGenerateInteractor implements PageGenerateInputBoundary {
+
+    final PageGenerateDsGateway pageGenerateDsGateway;
+    final PageGenerateOutputBoundary pageGenerateOutputBoundary;
+    final PageFactory pageFactory;
+
+    public PageGenerateInteractor(PageGenerateDsGateway pageGenerateDsGateway,
+                                  PageGenerateOutputBoundary pageGenerateOutputBoundary, PageFactory pageFactory){
+        this.pageGenerateDsGateway = pageGenerateDsGateway;
+        this.pageGenerateOutputBoundary = pageGenerateOutputBoundary;
+        this.pageFactory = pageFactory;
+    }
+
+    public PageGenerateResponseModel create(PageGenerateRequestModel requestModel){
+
+        if (pageGenerateDsGateway.existsByPageLabel(requestModel.getInputLabel())){
+            return pageGenerateOutputBoundary.prepareFailView("Page already exists.");
+        }
+
+        Page page = pageFactory.create(requestModel.getInputLabel(), requestModel.getUser());
+
+        PageGenerateResponseModel pageResponseModel;
+
+        if (requestModel.getPageType().equals("Corporate")){
+            CorporatePage corporatePage = (CorporatePage) page;
+            CorporatePageFactory corporatePageFactory = new CorporatePageFactory();
+            corporatePage = corporatePageFactory.addInfo(corporatePage,
+                    requestModel.getCompanyName(), requestModel.getCompanyInfo());
+
+            saveToDs(corporatePage);
+
+            pageResponseModel = new PageGenerateResponseModel(
+                    "New corporate page successfully created.",
+                    corporatePage.getCompanyName(), corporatePage.getCompanyInfo());
+
+        } else {
+            saveToDs(page);
+            pageResponseModel = new PageGenerateResponseModel(
+                    "page successfully created.");
+        }
+
+        return pageGenerateOutputBoundary.prepareSuccessView(pageResponseModel);
```

Comment:
Here since there is no "view" (we don't use Java for front-end), we can return a response model which will be sent back to the Javascript client.